common/Png.hpp

```diff
@@ -92,20 +92,28 @@ unpremultiply_data (png_structp /*png*/, png_row_infop row_info, png_bytep data)
     for (i = 0; i < row_info->rowbytes; i += 4)
     {
         uint8_t *b = &data[i];
-        uint32_t pixel;
+        uint32_t pix;
         uint8_t  alpha;
 
-        std::memcpy (&pixel, b, sizeof (uint32_t));
-        alpha = (pixel & 0xff000000) >> 24;
-        if (alpha == 0)
+        std::memcpy (&pix, b, sizeof (uint32_t));
+
+        alpha = (pix & 0xff000000) >> 24;
+        if (alpha == 255)
+        {
+            b[0] = ((pix & 0xff0000) >> 16);
+            b[1] = ((pix & 0x00ff00) >>  8);
+            b[2] = ((pix & 0x0000ff) >>  0);
```

Comment:
I would imagine 'shit and truncate' to be the more natural form, because the `&` is done way with in the `mov` byte instruction (it's only there to avoid warnings of size-mismatch implicit casting) ...

`b[0] = ((pix >> 16) & 0xff);`
`b[1] = ((pix >> 8) & 0xff);`
`b[2] = (pix & 0xff);`

Although I imagine a good compiler will generate the same code (but isn't guaranteed), anyway, technically this is the more explicit form for what the code is doing (copying a specific byte from the int) and it almost certainly will produce the most optimal code.

Notice that the code below is not using this form because we don't narrow from int to char. So the below code is correct.